[Build] Absl cuda warning on Linux

### Describe the issue

It looks like a similar issue to what https://github.com/microsoft/onnxruntime/pull/27096 might be able to fix.
It's only we need to apply the patch to linux as well?

https://github.com/microsoft/onnxruntime/blob/ec4f6bfa1f68ec7aee1608fb5ee3f7f0917330fe/cmake/external/abseil-cpp.cmake#L23-L28

### Urgency

_No response_

### Target platform

Linux

### Build script

./build.sh --config RelWithDebInfo --use_cuda --cuda_home /path/to/cuda12.8 --cudnn_home /path/to/cudnn9.8 --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES=90 --build_wheel --parallel --skip_tests


### Error / output

```
[ 51%] Building CUDA object CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/tensor/gelu_impl.cu.o
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1466: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/activation/activations_impl.cu.o] Error 255
make[2]: *** Waiting for unfinished jobs....
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/cuda_utils.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1481: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/cuda_utils.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/generator/random_impl.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1511: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/generator/random_impl.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/fpgeneric.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1496: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/fpgeneric.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/llm/rotary_embedding_impl.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1541: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/llm/rotary_embedding_impl.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/einsum_utils/einsum_auxiliary_ops_diagonal.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1601: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/einsum_utils/einsum_auxiliary_ops_diagonal.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/binary_elementwise_ops_impl.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1556: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/binary_elementwise_ops_impl.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/cumsum_impl.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1586: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/cumsum_impl.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/generator/range_impl.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1526: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/generator/range_impl.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/clip_impl.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1571: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/clip_impl.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/variadic_elementwise_ops_impl.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1736: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/variadic_elementwise_ops_impl.cu.o] Error 255
/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/hash/internal/hash.h(481): error #68-D: integer conversion resulted in a change of sign
        sizeof(T) == -1,
                     ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/home/titaiwang/onnxruntime/build/Linux/RelWithDebInfo/_deps/abseil_cpp-src/absl/container/internal/raw_hash_set.h(468): error #69-D: integer conversion resulted in truncation
        static_cast<uint16_t>(reinterpret_cast<uintptr_t>(&seed));
                    ^

2 errors detected in the compilation of "/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/matmul_integer.cu".
make[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1616: CMakeFiles/onnxruntime_providers_cuda.dir/home/titaiwang/onnxruntime/onnxruntime/core/providers/cuda/math/matmul_integer.cu.o] Error 255
```

### Visual Studio Version

_No response_

### GCC / Compiler Version

_No response_

--- a/cmake/external/abseil-cpp.cmake
+++ b/cmake/external/abseil-cpp.cmake
@@ -27,6 +27,8 @@ if(Patch_FOUND)
   else()
     set(ABSL_PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/abseil/absl_cuda_warnings.patch)
   endif()
+else()
+  set(ABSL_PATCH_COMMAND "")
 endif()
 
 # NB! Advancing Abseil version changes its internal namespace,

